zerver/actions/realm_linkifiers.py

```diff
@@ -131,3 +139,42 @@ def do_update_linkifier(
     )
 
     notify_linkifiers(realm, realm_linkifiers)
+
+
+@transaction.atomic(durable=True)
+def check_reorder_linkifiers(
+    realm: Realm, ordered_linkifier_ids: List[int], *, acting_user: Optional[UserProfile]
+) -> None:
+    """ordered_linkifier_ids should contain ids of all existing linkifiers.
+    For the rare situation when any of the linkifier gets deleted that more ids are passed,
+    we just ignore it because it has no effect on the ordering"""
+    id_to_new_order = {
+        linkifier_id: order for order, linkifier_id in enumerate(ordered_linkifier_ids)
+    }
+    # Reject repeated IDs in the user request
+    if len(id_to_new_order) < len(ordered_linkifier_ids):
+        raise JsonableError(_("The ordered list must not contain duplicated linkifiers"))
+    linkifiers = RealmFilter.objects.filter(realm=realm)
+    for linkifier in linkifiers:
+        if linkifier.id not in id_to_new_order:
+            raise JsonableError(
+                _("Linkifier with ID {linkifier_id} does not have a specified order").format(
+                    linkifier_id=linkifier.id
+                )
+            )
+        linkifier.order = id_to_new_order[linkifier.id]
+        linkifier.save(update_fields=["order"])
```

Comment:
It'd be nice to avoid doing database queries in a loop her -- this seems like a good application for `bulk_update`?

Reply:
Yeah that's doable. Just need to flush the cache because `flush_linkifiers` was connected only to `post_save` and `post_delete`.